app/containers/AudioPlayer/Seek.tsx

```diff
@@ -0,0 +1,151 @@
+import Animated, {
+	SharedValue,
+	runOnJS,
+	useAnimatedGestureHandler,
+	useAnimatedProps,
+	useAnimatedStyle,
+	useDerivedValue,
+	useSharedValue
+} from 'react-native-reanimated';
+
+import styles from './styles';
+import { useTheme } from '../../theme';
+
+const AnimatedTextInput = Animated.createAnimatedComponent(TextInput);
+
+interface ISeek {
+	duration: SharedValue<number>;
+	currentTime: SharedValue<number>;
+	loaded: boolean;
+	onChangeTime: (time: number) => Promise<void>;
+}
+
+const BUTTON_HIT_SLOP = {
+	top: 8,
+	right: 8,
+	bottom: 8,
+	left: 8
+};
+
+const Seek = ({ currentTime, duration, loaded = false, onChangeTime }: ISeek) => {
+	const { colors } = useTheme();
+
+	const maxWidth = useSharedValue(1);
+	const x = useSharedValue(0);
+	const current = useSharedValue('00:00');
+	const scale = useSharedValue(1);
+	const isHandlePan = useSharedValue(false);
+	const onEndGestureHandler = useSharedValue(false);
+	const isTimeChanged = useSharedValue(false);
+
+	const styleLine = useAnimatedStyle(() => ({
+		width: x.value,
+		zIndex: 2
+	}));
+
+	const styleThumb = useAnimatedStyle(() => ({
+		transform: [{ translateX: x.value }, { scale: scale.value }]
+	}));
+
+	const onLayout = (event: LayoutChangeEvent) => {
+		const { width } = event.nativeEvent.layout;
+		maxWidth.value = width - 12;
+	};
+
+	const gestureHandler = useAnimatedGestureHandler({
+		onStart: (_, ctx: any) => {
+			ctx.startX = x.value;
+			isHandlePan.value = true;
+		},
+		onActive: (event, ctx: any) => {
+			const moveInX: number = ctx.startX + event.translationX;
+			if (moveInX < 0) {
+				x.value = 0;
+			} else if (moveInX > maxWidth.value) {
+				x.value = maxWidth.value;
+			} else {
+				x.value = moveInX;
+			}
+			isTimeChanged.value = true;
+			scale.value = 1.3;
+		},
+		onEnd: () => {
+			scale.value = 1;
+			isHandlePan.value = false;
+			onEndGestureHandler.value = true;
+		}
+	});
+
+	const wrapper = async (time: number) => {
+		await onChangeTime(Math.round(time * 1000));
+		onEndGestureHandler.value = false;
+	};
+
+	useDerivedValue(() => {
+		if (onEndGestureHandler.value) {
+			runOnJS(wrapper)(currentTime.value);
+		}
+	});
+
+	// https://docs.swmansion.com/react-native-reanimated/docs/2.x/fundamentals/worklets/
+	const formatTime = (ms: number) => {
+		'worklet';
+
+		const minutes = Math.floor(ms / 60);
+		const remainingSeconds = Math.floor(ms % 60);
+		const formattedMinutes = String(minutes).padStart(2, '0');
+		const formattedSeconds = String(remainingSeconds).padStart(2, '0');
+		return `${formattedMinutes}:${formattedSeconds}`;
+	};
+
+	useDerivedValue(() => {
+		if (isHandlePan.value) {
+			const cTime = (x.value * duration.value) / maxWidth.value || 0;
+			currentTime.value = cTime;
+			current.value = formatTime(cTime);
+		} else {
+			const xTime = (currentTime.value * maxWidth.value) / duration.value || 0;
+			x.value = xTime;
+			current.value = formatTime(currentTime.value);
```

Comment:
Besides `cTime` and `xTime` naming, can you explain what's happening here?
They seem to do the same thing.

Reply:
I changed the name of variables, the cTime is timeSelected, xTime is timeInProgress, current is timeLabel and x is timePosition.

- The first one is when you hold the thumb icon and move it through the line changing his position. To calculate the time selected based on the position, then changes the currentTime that is the same time as the Audio. 
- The other is the timeInProgress, to calculate the position based on the time received from the Audio. Then changes the timePosition to move the seek to the correct position.
- Both changes the value of timeLabel(current), because they are changing the time label